client/db.go

```diff
@@ -423,9 +426,9 @@ func sendAndFill(send func(...proto.Call) error, b *Batch) error {
 // Upon completion, Batch.Results will contain the results for each
 // operation. The order of the results matches the order the operations were
 // added to the batch.
-func (db *DB) Run(b *Batch) error {
+func (db *DB) Run(b *Batch) *proto.Error {
```

Comment:
Rather than changing this return type, could `proto.Error` implement the `error` interface and let the caller extract the `proto.Error` type when desired?